pkg/scalers/pulsar_scaler.go

```diff
@@ -0,0 +1,301 @@
+	default:
+		return meta, errors.New("no subscription given")
+	}
+
+	meta.activationTargetQueryValue = 0
+	if val, ok := config.TriggerMetadata["activationTargetQueryValue"]; ok {
+		activationTargetQueryValue, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return meta, fmt.Errorf("activationTargetQueryValue parsing error %s", err.Error())
+		}
+		meta.activationTargetQueryValue = activationTargetQueryValue
+	}
+
+	meta.msgBacklogThreshold = defaultMsgBacklogThreshold
+
+	if val, ok := config.TriggerMetadata[msgBacklogMetricName]; ok {
+		t, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return meta, fmt.Errorf("error parsing %s: %s", msgBacklogMetricName, err)
+		}
+		meta.msgBacklogThreshold = t
+	}
+
+	meta.enableTLS = false
+	if val, ok := config.TriggerMetadata["tls"]; ok {
+		val = strings.TrimSpace(val)
+
+		if val == enable {
+			cert := config.AuthParams["cert"]
+			key := config.AuthParams["key"]
+			if key == "" || cert == "" {
+				return meta, errors.New("must be provided cert and key")
+			}
+			meta.ca = config.AuthParams["ca"]
+			meta.cert = cert
+			meta.key = key
+			meta.enableTLS = true
+		} else {
+			return meta, fmt.Errorf("err incorrect value for TLS given: %s", val)
+		}
+	}
+	meta.scalerIndex = config.ScalerIndex
+	return meta, nil
+}
+
+func (s *pulsarScaler) GetStats(ctx context.Context) (*pulsarStats, error) {
+	stats := new(pulsarStats)
+
+	req, err := http.NewRequestWithContext(ctx, "GET", s.metadata.statsURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error requesting stats from url: %s", err)
+	}
+
+	res, err := s.client.Do(req)
+	if res == nil || err != nil {
+		return nil, fmt.Errorf("error requesting stats from url: %s", err)
+	}
+
+	defer res.Body.Close()
+
+	switch res.StatusCode {
+	case 200:
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return nil, fmt.Errorf("error requesting stats from url: %s", err)
+		}
+		err = json.Unmarshal(body, stats)
+		if err != nil {
+			return nil, fmt.Errorf("error unmarshalling response: %s", err)
+		}
+		return stats, nil
+	case 404:
+		return nil, nil
+	default:
+		return nil, fmt.Errorf("error requesting stats from url: %s", err)
+	}
+}
+
+func (s *pulsarScaler) getMsgBackLog(ctx context.Context) (int64, bool, error) {
+	stats, err := s.GetStats(ctx)
+	if err != nil {
+		return 0, false, err
+	}
+
+	if stats == nil {
+		return 0, false, nil
+	}
+
+	v, found := stats.Subscriptions[s.metadata.subscription]
+
+	return v.Msgbacklog, found, nil
+}
+
+// IsActive determines if we need to scale from zero
+func (s *pulsarScaler) IsActive(ctx context.Context) (bool, error) {
+	msgBackLog, found, err := s.getMsgBackLog(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	if !found || msgBackLog <= s.metadata.activationTargetQueryValue {
+		pulsarLog.Info("Pulsar subscription is not active, either no subscription found or no backlog detected")
+		return false, nil
+	}
+
+	return true, nil
+}
```

Comment:
Maybe instead of this approach, we could unify and simplify the logic doing something like:
```suggestion
	if !found {
		pulsarLog.Info("Pulsar subscription is not active, either no subscription found or no backlog detected")
		return false, nil
	}

	return msgBackLog > s.metadata.activationTargetQueryValue, nil
}
```